src/main/webapp/app/scores/exercise-scores.component.ts

```diff
@@ -163,8 +168,18 @@ export class ExerciseScoresComponent implements OnInit, OnDestroy {
     }
 
     goToBuildPlan(result: Result) {
-        // TODO: get the continuous integration URL as a client constant during the management info call
-        window.open('https://bamboobruegge.in.tum.de/browse/' + (result.participation! as ProgrammingExerciseStudentParticipation).buildPlanId);
+        const planId = (result.participation! as ProgrammingExerciseStudentParticipation).buildPlanId;
+        const projectKey = (this.exercise as ProgrammingExercise).projectKey!;
+        this.profileService
```

Comment:
This looks really really complicated. Can we not achieve this in an easier way?

Reply:
I moved the code to a new directive `build-plan-button.directive.ts`, so that the components only have to provide the projectKey and buildPlanId.